docs/init-edge.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -e
+
+K8S_NODE_NAME=local-edge-control-plane
+SPIRE_AGENT_HOSTNAME=$(hostname)-local-edge
+
+# Create spire agent's join token
+SPIRE_AGENT_JOIN_TOKEN=$(kubectl exec -n spire spire-server-0 -c spire-server -- \
+  /opt/spire/bin/spire-server token generate -spiffeID "spiffe://katulu.io/$SPIRE_AGENT_HOSTNAME" -ttl 3600 | \
+  awk '{print $2}')
+
+# Allow the local-edge kind cluster's fl-clients to communicate with the fl-server
+kubectl exec -n spire spire-server-0 -c spire-server -- /opt/spire/bin/spire-server entry create \
+    -parentID "spiffe://katulu.io/$SPIRE_AGENT_HOSTNAME" \
+    -spiffeID spiffe://katulu.io/flower-client \
+    -selector "k8s:node-name:$K8S_NODE_NAME" \
+    -selector "k8s:ns:katulu-fl" \
+    -selector "k8s:pod-label:app:flower-client"
+# Allow the local-edge kind cluster's fl-operator to communicate with the fl-orchestrator
+kubectl exec -n spire spire-server-0 -c spire-server -- /opt/spire/bin/spire-server entry create \
+    -parentID "spiffe://katulu.io/$SPIRE_AGENT_HOSTNAME" \
+    -spiffeID spiffe://katulu.io/fl-operator \
+    -selector "k8s:node-name:$K8S_NODE_NAME" \
+    -selector "k8s:ns:katulu-fl" \
+    -selector "k8s:pod-label:app:fl-operator-envoyproxy"
+
+# Setup spire agent to communicate with the spire server
+SPIRE_AGENT_SERVER_ADDRESS=$(kubectl -n spire get virtualservice spire-server -o jsonpath='{ .spec.hosts[0] }')
+cat <<EOF > kustomize/fl-edge/config/spire-server-creds.env
+spire_agent_join_token=$SPIRE_AGENT_JOIN_TOKEN
+spire_agent_server_address=$SPIRE_AGENT_SERVER_ADDRESS
+EOF
+
+# Setup fl-operator to communicate with the fl-orchestrator
+FL_ORCHESTRATOR_SNI=$(kubectl -n katulu-fl get virtualservice fl-orchestrator-envoyproxy -o jsonpath='{ .spec.hosts[0] }')
+cat <<EOF > kustomize/fl-edge/config/fl-operator-params.env
+fl_orchestrator_sni=$FL_ORCHESTRATOR_SNI
+fl_orchestrator_port=8443
+EOF
+
+# Deploy kind cluster that uses fl-studio's container registry
+CONTAINER_REGISTRY_FQDN=$(kubectl -n container-registry get virtualservice container-registry -o jsonpath='{ .spec.hosts[0] }')
+export CONTAINER_REGISTRY_FQDN
+envsubst <<EOF | kind create cluster --kubeconfig local-edge-kubeconfig.yaml --config -
```

Comment:
Is it important to use this specific `kubeconfig.yaml`? Wouldn't kind create a new one anyways?

Reply:
Kind by default uses `$HOME/.kube/config`. And we want to use a different `kubeconfig.yaml` so that the `port-forward` in the *Access the fl-suite* stage is unaltered.